src/ut/httpstack_test.cpp

```diff
@@ -341,6 +356,61 @@ TEST_F(HttpStackTest, SASCorrelationHeader)
   mock_sas_collect_messages(false);
 }
 
+// Check that the ProxiedSasLogger picks up X-Real-IP and X-Real-Port headers
+TEST_F(HttpStackTest, RealIPHeader)
+{
+  mock_sas_collect_messages(true);
+  start_stack();
+
+  ProxiedHandler handler;
+  _stack->register_handler("^/ProxiedHandler$", &handler);
+
+  int status;
+  std::string response;
+  std::list<std::string> hdrs;
+  hdrs.push_back("X-Real-Ip: 12.34.56.78");
+  hdrs.push_back("X-Real-Port: 4242");
+
+  int rc = get("/ProxiedHandler", status, response, &hdrs);
+  ASSERT_EQ(CURLE_OK, rc);
+  ASSERT_EQ(200, status);
+  ASSERT_EQ("OK", response);
+
+  MockSASMessage* message = mock_sas_find_event(SASEvent::RX_HTTP_REQ);
+  EXPECT_TRUE(message != NULL);
+  EXPECT_EQ(message->var_params[0], "12.34.56.78");
+  EXPECT_EQ(message->static_params[0], 4242);
+
+  stop_stack();
+  mock_sas_collect_messages(false);
+}
+
+// Check that the ProxiedSasLogger picks logs real IPs if not X-Real-Ip header is present
```

Comment:
*Check that the ProxiedSasLogger logs the real IPs if the X-Real-Ip header is not present